pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -94,16 +106,12 @@ func newRepositoryWithClient(registryClient client.Interface, quotaClient kclien
 		return nil, errors.New("DOCKER_REGISTRY_URL is required")
 	}
 
-	pullthrough := false
-	if value, ok := options["pullthrough"]; ok {
-		if b, ok := value.(bool); ok {
-			pullthrough = b
-		}
-	}
+	pullthrough := getBoolOption("pullthrough", false, options)
+	allowSchema2 := getBoolOption("allowSchema2", false, options)
```

Comment:
Can the option be renamed to `acceptschema2`? I'd make it lower-case in any case.